app/models/concerns/availability_status_concern.rb

```diff
@@ -0,0 +1,25 @@
+module AvailabilityStatusConcern
+  extend ActiveSupport::Concern
+
+  included do
+    before_update :update_status
+  end
+
+  private
+
+  STATUS_RELATED_ATTRS = [
+    "availability_status",
+    "last_checked_at",
+    "updated_at",
+    "last_available_at",
+    "availability_status_error"
+  ].freeze
+
+  def update_status
```

Comment:
I'm wondering whether the availability status update doesn't cause availability status reset by this code?